test/components/preprocessors/test_markdown_header_splitter.py

```diff
@@ -0,0 +1,252 @@
+
+
+def test_empty_document_list():
+    """Test handling of an empty document list."""
+    splitter = MarkdownHeaderSplitter()
+    result = splitter.run(documents=[])
+    assert result["documents"] == []
+
+
+def test_invalid_secondary_split():
+    """Test that an invalid secondary split type raises an error."""
+    # In MarkdownHeaderSplitter, this is validated at DocumentSplitter instantiation time in _apply_secondary_splitting
+    splitter = MarkdownHeaderSplitter(secondary_split="invalid_split_type")
+    docs = [Document(content="# Header\nContent")]
+
+    # Error should be raised when run is called and secondary splitter is created
+    with pytest.raises(ValueError, match="split_by must be one of"):
+        splitter.run(documents=docs)
+
+
+def test_invalid_split_parameters():
+    """Test invalid split parameter validation."""
+    # Similar to invalid_secondary_split, validation happens at DocumentSplitter instantiation
+
+    # Test split_length validation
+    splitter = MarkdownHeaderSplitter(secondary_split="word", split_length=0)
+    docs = [Document(content="# Header\nContent")]
+    with pytest.raises(ValueError, match="split_length must be greater than 0"):
+        splitter.run(documents=docs)
+
+    # Test split_overlap validation
+    splitter = MarkdownHeaderSplitter(secondary_split="word", split_overlap=-1)
+    docs = [Document(content="# Header\nContent")]
+    with pytest.raises(ValueError, match="split_overlap must be greater than or equal to 0"):
+        splitter.run(documents=docs)
+
+
+def test_empty_content_handling():
+    """Test handling of documents with empty content."""
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content="")]
+    result = splitter.run(documents=docs)
+
+    # DocumentSplitter skips empty documents by default
+    assert len(result["documents"]) == 0
+
+
+# Output format and split ID checks
+def test_document_splitting_format():
+    """Test that the format of split documents is correct."""
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content="# Header\nContent")]
+    result = splitter.run(documents=docs)
+
+    # Basic validation of the output structure
+    assert isinstance(result, dict)
+    assert "documents" in result
+    assert isinstance(result["documents"], list)
+
+
+def test_split_id_sequentiality_primary_and_secondary():
+    text = "# Header\n" + "Word " * 30
+    # Test primary splitting
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content=text)]
+    result = splitter.run(documents=docs)
+    split_ids = [doc.meta["split_id"] for doc in result["documents"]]
+    assert split_ids == list(range(len(split_ids)))
+
+    # Test secondary splitting
+    splitter = MarkdownHeaderSplitter(secondary_split="word", split_length=5)
+    result = splitter.run(documents=docs)
+    split_ids = [doc.meta["split_id"] for doc in result["documents"]]
+    assert split_ids == list(range(len(split_ids)))
```

Comment:
Could we add additional tests that include examples using `split_overlap`, `split_threshold`, and examples that use page breaks in them to test if the page break functionality works as expected?